### DISCUSSION TITLE
Using JavaScript code in the browser version

### DISCUSSION THREAD

**Discussion by IVANKIRICHENKO04** [AvaloniaUI/Avalonia#15379] [Q&A]
Are there any ready-made solutions for executing JavaScript code from the browser version of Avalonia? I need to open a file manager to upload an image, but I can't do it directly from the browser. How can this be fixed?
Thanks in advance.

**Comment by timunie** [comment]
TopLevel.StorageProvider isn't usable in Browser?

**Reply by maxkatz6** [reply]
It should be.

**Comment by maxkatz6** [comment]
Use JSImport/JSExport https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0

**Comment by IVANKIRICHENKO04** [comment]
Thanks for the help!
